tox.ini

```diff
@@ -19,3 +19,28 @@ deps =
     yamllint==1.15.0
 commands =
     bash -c "yamllint eve/main.yml salt/**/*.yaml"
+
+[testenv:ci-tests]
+deps =
+    pytest
+    pytest_bdd
+    testinfra
+    paramiko
+    cryptography==2.4.2
+commands =
+    pytest --ssh-config=some/identity/files --hosts=[list, of, hosts] -k "ci" tests
+
+[testenv:local-tests]
+setenv =
+    VAGRANT_CWD = {toxinidir}
+commands_pre =
+    bash -c "vagrant up"
+    bash -c "vagrant ssh-config >{toxinidir}/ssh.config"
+deps =
+    pytest
+    pytest_bdd
+    testinfra
+    paramiko
+    cryptography==2.4.2
+commands =
+    pytest --ssh-config={toxinidir}/ssh.config --hosts="bootstrap" -k "local" tests
```

Comment:
You can instead have `{posargs:-k "local"}` and a single target, so we default to local tests and can specify overrides if we wish (in the CI command for example).

Reply:
what about vagrant stuff ?

Reply:
2 separate concerns for me:

* the annotations on the gherkin E2E scenarios should have a useful default (with a sensible RTT)
* the test suite should be able to manage machines 2 different ways depending on a `CI_ENV` flag or something

The intent being to have a configurable E2E target in terms of the amount and objectives of its tests, but have it manage both platforms equally within. I'm not sure this is possible with a single target (in particular w.r.t. how you set the credentials for the initial machine, with Vagrant/OpenStack differences), but to me, you should have a sensible default and allow overriding rather than setting different annotation targets in each one.